allensdk/brain_observatory/behavior/behavior_project_cache/tables/util/metadata_parsers.py

```diff
@@ -0,0 +1,245 @@
+    elif split_str[2] == "gratings":
+        stim_set = "gratings"
+    return stim_set
+
+
+def parse_stimulus_set_novelty(metadata_row: pd.Series) -> Union[str, None]:
+    """Return familiarity with stimulus for session.
+
+    Parameters
+    ----------
+    metadata_row : pandas.Series
+        Complete set of metadata for session.
+
+    Returns
+    -------
+    stimulus_set_novelty : str
+        ``familiar`` or ``novel`` for session.
+    """
+    familiar_ophys_sessions = ["0", "1", "2", "3"]
+
+    session_type = metadata_row["session_type"]
+    if session_type not in SESSION_TYPES:
+        return None
+
+    if "TRAINING" in session_type:
+        stimulus_set_novelty = "training"
+    elif "7" in session_type:
+        stimulus_set_novelty = "NA"
+    else:
+        is_familiar = any(
+            familiar_num in session_type
+            for familiar_num in familiar_ophys_sessions
+        )
+        if is_familiar:
+            stimulus_set_novelty = "familiar"
+        else:
+            stimulus_set_novelty = "novel"
+    return stimulus_set_novelty
+
+
+############################
+#
+#     from project code
+#
+############################
+
+
+def parse_num_cortical_structures(metadata_row: pd.Series) -> Union[str, None]:
+    """Return the number of structures that were targeted for this session.
+
+    Parameters
+    ----------
+    metadata_row : pandas.Series
+        Complete set of metadata for session.
+
+    Returns
+    -------
+    num_structures : int
+        Number of structures targeted for the session.
+    """
+    attr_names = dir(metadata_row)
+    if "project_code" in attr_names:
+        project_code = metadata_row["project_code"]
+    elif "project_code_behavior" in attr_names:
+        project_code = metadata_row["project_code_behavior"]
+    elif "project_code_ophys" in attr_names:
+        project_code = metadata_row["project_code_ophys"]
+    else:
+        return None
+
+    if project_code not in PROJECT_CODES:
+        return None
+    num_structures = NUM_STRUCTURES_DICT[project_code]
+    return num_structures
+
+
+def parse_num_depths(metadata_row: pd.Series) -> Union[str, None]:
+    """Return the number of depths that were imaged for the session.
+
+    Parameters
+    ----------
+    metadata_row : pandas.Series
+        Complete set of metadata for session.
+
+    Returns
+    -------
+    num_depths : int
+        Number of depths imaged for the session.
+    """
+    attr_names = dir(metadata_row)
+    if "project_code" in attr_names:
+        project_code = metadata_row["project_code"]
+    elif "project_code_behavior" in attr_names:
+        project_code = metadata_row["project_code_behavior"]
+    elif "project_code_ophys" in attr_names:
+        project_code = metadata_row["project_code_ophys"]
+    else:
+        return None
+
+    if project_code not in PROJECT_CODES:
+        return None
+    num_depths = NUM_DEPTHS_DICT[project_code]
```

Comment:
better to do `num_depths = NUM_DEPTHS_DICT.get(project_code, None)` rather than checking a separate list which value may not be in `NUM_DEPTHS_DICT`